ndcube/asdf/converters/globalcoords_converter.py

```diff
@@ -0,0 +1,26 @@
+from typing import OrderedDict
+
+from asdf.extension import Converter
+
+
+class GlobalCoordsConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/global_coords/globalcoords-*"]
+    types = ["ndcube.global_coords.GlobalCoords"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.global_coords import GlobalCoords
+
+        globalcoords = GlobalCoords()
+        if node.get("internal_coords") is not None:
+            globalcoords._internal_coords = OrderedDict(node.get("internal_coords"))
```

Comment:
```suggestion
        if "internal_coords" in node:
            globalcoords._internal_coords = node["internal_coords"]
```
See follow-up comment about `OrderedDict`

Reply:
Sure